core/src/main/java/org/bouncycastle/crypto/engines/RFC5649WrapEngine.java

```diff
@@ -215,12 +215,13 @@ public String getAlgorithmName()
             isValid = false;
         }
 
-        // Check the number of padded zeros
+        // Check the number of padding zeros
         int expectedZeros = upperBound - mli;
-        if (expectedZeros >= paddedPlaintext.length)
+        if (expectedZeros >= 8 || expectedZeros < 0)
         {
+            // We have to pick a "typical" amount of padding to avoid timing attacks.
             isValid = false;
-            expectedZeros = paddedPlaintext.length;
+            expectedZeros = 4;
```

Comment:
I believe this was a timing attack. The expected number of padding zeros was from 0 to 7. By performing a verification of the entire plain text we were doing a lot more work on the comparison than is typical.